core/unit_test/TestMDRangePolicyConstructors.hpp

```diff
@@ -93,6 +93,19 @@ TEST(TEST_CATEGORY_DEATH, policy_bounds_unsafe_narrowing_conversions) {
       },
       "unsafe narrowing conversion");
 }
+
+TEST(TEST_CATEGORY_DEATH, policy_invalid_bounds) {
+  using Policy = Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<2>>;
+
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+  ASSERT_DEATH(
+      {
+        (void)Policy({100, 100}, {90, 90});
+      },
+      "Kokkos::MDRangePolicy bounds error: The lower bound \\(100\\) is "
+      "greater "
+      "than its upper bound \\(90\\) in dimension 1\\.");
```

Comment:
I hope the test fails...

Reply:
The failing dimension depends if you use `Iterate::Left` or `Iterate::Right`

Reply:
So we expect the test as is should not pass on all backends?

Reply:
Yes, the test should pass for some backends and fail for others

Reply:
Have we decided anyway that the bounds check should always produce an error?

Reply:
@dalg24 @Rombur is correct here. It's `Iterate::Right` by default here. So the check would start from `rank-1` dimension first.
https://github.com/kokkos/kokkos/blob/c3dde624d4228b1dd7f8952bc2439a64db5207ee/core/src/KokkosExp_MDRangePolicy.hpp#L318-L327

@masterleinad No, we haven't decided on that. If we do an abort here for `MDRangePolicy`, it's inconsistent from how `RangePolicy` handles this, so I think we definitely should discuss at some point before merging.

Reply:
I would vote to abort and also change range policy to reflect this for the same reason of what I said above.

Reply:
Fixed to check against the policy's iteration pattern.

> I would vote to abort and also change range policy to reflect this for the same reason of what I said in the other comment

I agree with this approach as well. My only concern is if that will cause many breakages for the users. Although the change is for a corner case, it's a sudden behavioral change in a commonly used function.